### ISSUE TITLE
Expose additional parameters for `lab generate` and `lab train` to end users

### ISSUE THREAD

**Issue by jaideepr97** [instructlab/instructlab#379]
There are a few parameters we should consider exposing to end users in order to enable them to improve the quality of their models' outputs

This list is not exhaustive to the follwing:
`--num-examples` in `lab generate` to allow users to dictate how many of their provided examples are used to generate synthetic data 

`--learning-rate` and `--batch-size` to influence how the model trains in `lab train`

any others folks can think of

**Comment by khaledsulayman** [comment]
@mbestavros has agreed to take this on, though I'm currently unable to assign him.

**Comment by nathan-weinberg** [comment]
@cdoern @mbestavros any update on this?

**Comment by mbestavros** [comment]
I did some exploratory work here, but ended up pivoting to other stuff (signing, mainly). It also seemed like there was going to be a lot of overlap with @cdoern's redesign PRs.

**Comment by ktam3** [comment]
can someone confirm if this is GA blocker? I moved it to .19 but if this belong is .18 lmk

**Comment by jaideepr97** [comment]
@Maxusmusti @JamesKunstle @RobotSail 
This is a really old issue but I want to check with yall to see if there are any new training args/parameters we might want to make configurable to keep this around? 
I see learning rate for one is still not a field in the config.yaml - is that something (among other things) that should be exposed to users?

**Comment by jaideepr97** [comment]
> @jaideepr97 this was already implemented by @cdoern with his `additional_args` config field. He can validate whether or not this is still a relevant issue.

Thanks oleg
@cdoern what do you think? I guess that opens up the question about whether any of the args passed through additional args should be promoted to be first class args in the config/get a flag of their own?
if not I would propose closing this issue

**Comment by jaideepr97** [comment]
for reference: https://github.com/instructlab/instructlab/pull/1634

**Comment by ktam3** [comment]
outdated
